### PR TITLE
Load demo data with accessor

### LINKED ISSUE #546: Change retail demo to initialize Accessor
- `load_retail` currently returns a DataTable with the option to return just the DataFrame, but we should have it such that it returns the DataFrame with the Woodwork Accessor initialized by default and just the DataFrame if specified.

#### Code Example

```python
df = load_retail()

assert df.ww.schema is not None
```

### PR DESCRIPTION
**Pull request by tamargrey**
- Adds `load_retail_to_accessor` to allow loading the demo dataset in a way that returns a DataFrame with the Woodwork accessor initialized
- Closes #546

### CODE REVIEW

File: woodwork/demo/retail.py [hunk 1]
```diff
@@ -65,3 +65,56 @@ def load_retail(id='demo_retail_data', nrows=None, return_dataframe=False):
                       logical_types=logical_types)
 
     return dt
+
+
+def load_retail_to_accessor(id='demo_retail_data', nrows=None, init_woodwork=True):
+    """Load a demo retail dataset into a DataFrame, optionally initializing Woodwork's typing information.
+
+    Args:
+        id (str, optional): The name to assign to the DataTable, if returning a DataTable.
```
Comment (gsheni) [inline woodwork/demo/retail.py:74]: DataTable? I think you mean DataFrame?

File: woodwork/demo/retail.py [hunk 2]
```diff
@@ -65,3 +65,56 @@ def load_retail(id='demo_retail_data', nrows=None, return_dataframe=False):
...
+            If not returning a DataTable, this will be ignored. Defaults to ``demo_retail_data``.
+        nrows (int, optional): The number of rows to return in the dataset. If None, will
+            return all possible rows. Defaults to None.
+        init_woodwork (bool): If True, will return a pandas DataFrame with Woodwork
+        typing information initialized. If False, will return a DataFrame without
+        Woodwork initialized. Defaults to False.
+
+    Returns:
+         pd.DataFrame or pd.DataFrame + Woodwork: A DataFrame containing the demo data.
```
Comment (gsheni) [inline woodwork/demo/retail.py:83]: I would put just 
```suggestion
         pd.DataFrame: A DataFrame containing the demo data with Woodwork typing initialized. If `init_woodwork` is False, will return an uninitialized DataFrame. 
```

File: woodwork/demo/api.py [hunk 3]
```diff
@@ -1,2 +1,2 @@
 # flake8: noqa
-from .retail import load_retail
+from .retail import load_retail, load_retail_to_accessor
```
Comment (gsheni) [inline woodwork/demo/api.py:2]: Created follow up issue:
- https://github.com/alteryx/woodwork/issues/557

**Review by gsheni — COMMENTED** [review]
looks good overall. Just minor docstring things.

File: woodwork/demo/retail.py (hunk 1 above)
Comment (tamargrey) [reply woodwork/demo/retail.py:74]: yep!

**Review by gsheni — APPROVED** [review]
lgtm. Just need to lint

**Pull request merged**